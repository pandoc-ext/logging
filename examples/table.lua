local utils = require 'pandoc.utils'

local logging = require 'logging'

function Table(table)
    local caption = table.caption
    if caption then
        logging.temp('caption', utils.stringify(caption))
    end

    local head = table.head
    for i, row in ipairs(head.rows) do
        for j, cell in ipairs(row.cells) do
            logging.temp(string.format('head.row[%d].cell[%d]', i, j), cell)
        end
    end
end
