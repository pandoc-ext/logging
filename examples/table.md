: Cats

Name   | Description
-------|------------
Tom    | Black and white
