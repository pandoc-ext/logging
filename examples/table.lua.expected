(#) caption Cats
(#) head.row[1].cell[1] pandoc.Cell {
  alignment: "AlignDefault"
  attr: Attr {
    attributes: AttributeList {}
    classes: List {}
    identifier: ""
  }
  col_span: 1
  contents: Blocks[1] {
    [1] Plain {
      content: Inlines[1] {
        [1] Str "Name"
      }
    }
  }
  row_span: 1
}
(#) head.row[1].cell[2] pandoc.Cell {
  alignment: "AlignDefault"
  attr: Attr {
    attributes: AttributeList {}
    classes: List {}
    identifier: ""
  }
  col_span: 1
  contents: Blocks[1] {
    [1] Plain {
      content: Inlines[1] {
        [1] Str "Description"
      }
    }
  }
  row_span: 1
}
